src/main/io/dataEdition.c

```diff
@@ -0,0 +1,1359 @@
+static uint8_t	selectedPage	= SELECTION_PAGE;
+static uint8_t	selectedLine	= 1;
+static int8_t	editionLine		= -1;
+static int8_t	selectedItem 	= -1;
+
+static bool shouldInitialiseThePage 	= true;
+static bool onGoingEditionProcess 		= false;
+
+static char tempString[10];
+
+extern const clivalue_t valueTable[];
+extern uint8_t numberOfRecordsInCLITable ;
+static uint8_t numberOfElementsInList;
+static int32_t valueInteger = 0;
+static float valueFloat = 0;
+
+#if (DATA_EDITION_FULL_VERSION)
+extern uint16_t rssi;
+static uint32_t lastAlarmSoundTime;
+static bool isAlarmSoundOn 		= true;
+extern uint8_t numberOfBaudrateValues ;
+extern uint8_t motorCount;
+extern uint8_t servoCount;
+#endif
+
+//======================================  Utilities  =======================================
+void initialiseOledDisplayPage(void)
+{
+    memset(&structureOfPage, 0, sizeof(structureOfPage));
+}
+
+void initDataEditionStructures(void)    // called in main
+{
+#ifdef DISPLAY
+    if (featureConfigured(FEATURE_DISPLAY)) {
+    initialiseOledDisplayPage();
+    }
+#endif
+	onGoingDataEditionWithSticks = false;
+}
+
+static void initializeEditionModeSpace(void)
+{
+	for (int i = 0; i < DEFAULT_DISPLAY_LINES; i++) {
+		structureOfPage.displayLine[i].isActive = false;
+		structureOfPage.displayLine[i].itemsOnLine = 1;
+	}
+	for (int i = 0; i < MAX_OF_EDITABLE_ITEMS; i++)
+		structureOfPage.editField[i].alive = false;
+
+	selectedLine = 1;
+	editionLine = -1;
+    selectedItem = -1;
+    shouldInitialiseThePage = false;
+	onGoingEditionProcess = false;
+	numberOfSecondaryPagesForThatPage = 0;
+}
+
+static uint8_t firstItemOnLine(uint8_t line)
+{
+    return line * THREE_ITEMS_PER_LINE;
+}
+
+static void highlightSelectedItemOnOledDisplay(void)
+{
+	switch (structureOfPage.editField[selectedItem].color){
+		case PIXEL_MIXED :
+			oledCharFormat = INVERSE_CHAR_FORMAT;
+			i2c_OLED_set_xy(positionInLineOfInvertedChar, editionLine);
+			i2c_OLED_send_char(contentOfPage.lineOfText[editionLine][positionInLineOfInvertedChar]);
+			break;
+		case PIXEL_INVERTED :
+			if (structureOfPage.editField[selectedItem].color == PIXEL_INVERTED) {
+				oledCharFormat = INVERSE_CHAR_FORMAT;
+				for(int i = structureOfPage.editField[selectedItem].leftPosition; i <= structureOfPage.editField[selectedItem].rightPosition; i++){
+					i2c_OLED_set_xy(i, editionLine);
+					i2c_OLED_send_char(contentOfPage.lineOfText[editionLine][i]);
+				}
+			}
+			break;
+	}
+	oledCharFormat = NORMAL_CHAR_FORMAT;
+}
+
+static void completeStringWithSpaces(uint8_t lineIndex)
+{
+    uint8_t length = strlen(contentOfPage.lineOfText[lineIndex]);
+
+    while (length < sizeof(contentOfPage.lineOfText[lineIndex]))
+    	contentOfPage.lineOfText[lineIndex][length++] = 0x20;
+}
+
+
+#if (DATA_EDITION_FULL_VERSION)
+static bool shouldTriggerAlarmNow(void)
+{
+	return ((millis() - lastAlarmSoundTime) >= (uint32_t)(masterConfig.telemetryConfig.hottAlarmSoundInterval * MILLISECONDS_IN_A_SECOND));
```

Comment:
it is better to use signed compare (unsigned comparison will fail (once) after 49 days ;-) )
there is `cmp16`/`cmp32` macros in CF  (that should be renamed to something more descriptive)